lib/src/Base/Graph/openturns/GridLayout.hxx

```diff
@@ -0,0 +1,88 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Grid layout of graphs
+ *
+ *  Copyright 2005-2020 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_GRIDLAYOUT_HXX
+#define OPENTURNS_GRIDLAYOUT_HXX
+
+#include "openturns/OTprivate.hxx"
+#include "openturns/GraphImplementation.hxx"
+#include "openturns/Graph.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+
+
+/**
+ * @class GridLayout
+ *
+ * The class GridLayout
+ */
+class OT_API GridLayout : public GraphImplementation
```

Comment:
What is the rational behind this inheritance? IMO GridLayout is essentially a matrix of graphs with some additional services, so it should be implemented this way. I may have missed something but I don't understand the interest of the proposed architecture and eg the isComposite() method.

Reply:
My point of view is that GridLayout can be considered as a Graph as it has some similar services : titles, accessors for the properties of all subgraphs (setAxes/setLegendPosition/setLogScale/setGrid/setGridColor)
Also I have to point out that not inheriting from Graph will not allow to return either a regular Graph or a GridLayout with the same prototype (covariant return types).

Reply:
Thanks for the explanations. I am not very fond of lists allowing to change the properties of its elements by reproducing their methods. You replace `myList[i].setProperty(property)` by `myList.setProperty(i, property)`, which is not a gain. If you introduce `myList.setProperties(propertyList)` it is a different method with a different name.
Concerning the covariant return types, a `draw` method can return either a Graph or a GridLayout depending on the class and the context. I can't see a situation where the base class would return a Graph and the derived class a GridLayout.
What bothers me the most is the introduction of the `isComposite()` method. It is a way to make a confusion between a set and one of its elements. It is an error from a mathematical point of view (a set is **always** different from one of its elements) so I can't imagine it is a good thing from a code pespective.
Do you have a specific use-case where GridLayout and Graph should be on the same level?

Reply:
I dont actually need GridLayout to belong to the Graph tree, so yes it can be moved.